### PR TITLE
JIRA #CORE-91; #CORE-92 Select Scala and SBT versions

### PR DESCRIPTION
**Pull request by DoSOfRedRiver**
I briefly checked that all projects work properly with sbt 1.0.3, please let me know if you have any suggestions about what can be broken by new sbt.
Scala 2.12.4 should not affect our codebase in any way.

### CODE REVIEW

File: rholang/project/BNFC.scala [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import sbt._
 import Keys._
+import scala.sys.process._
```
Comment (dckc) [inline rholang/project/BNFC.scala]: is this needed to get the code to compile with scala 2.12.4?

File: rholang/project/BNFC.scala [hunk 2]
```diff
@@ -50,8 +51,11 @@ object BNFC {
     bnfcOutputDir  := (javaSource in Compile).value,
     bnfcDocDir     := baseDirectory.value / "doc" / "bnfc",
     clean          := cleanDir(bnfcOutputDir.value / nsToPath(bnfcNamespace.value)),
-    generate       := bnfcFiles(bnfcGrammarDir.value).foreach { (f: File) =>
-      bnfcGenerateSources((fullClasspath in BNFCConfig).value, f, bnfcOutputDir.value, bnfcNamespace.value)
+    generate       := {
+      val fullCP = (fullClasspath in BNFCConfig).value
+      bnfcFiles(bnfcGrammarDir.value).foreach { (f: File) =>
+        bnfcGenerateSources(fullCP, f, bnfcOutputDir.value, bnfcNamespace.value)
+      }
```
Comment (dckc) [inline rholang/project/BNFC.scala]: likewise, is this needed to get the code to compile with scala 2.12.4?

File: rholang/project/BNFC.scala (hunk 1 above)
Comment (DoSOfRedRiver) [reply rholang/project/BNFC.scala]: Yep, I guess in sbt 1.x you need to explicitly import `process` package members to use.

File: rholang/project/BNFC.scala (hunk 2 above)
Comment (DoSOfRedRiver) [reply rholang/project/BNFC.scala]: This is another requirement from sbt 1.x for task invocation inside anonymous functions.

**Comment by DoSOfRedRiver** [comment]
@kirkwood Sorry for that. Hope that in near future we will have all necessary plugins in 1.x

**Pull request merged**